components/checkbox/src/index.tsx

```diff
@@ -9,12 +9,14 @@ import styled from 'styled-components';
 import HintText from '@govuk-react/hint-text';
 import { FOCUS_COLOUR } from 'govuk-colours';
 import { BORDER_WIDTH, BORDER_WIDTH_FORM_ELEMENT, FOCUS_WIDTH, SPACING_POINTS } from '@govuk-react/constants';
-import { spacing, typography } from '@govuk-react/lib';
+import { spacing, typography, WithWhiteSpaceProps } from '@govuk-react/lib';
+import type { Size } from '@govuk-react/lib';
```

Comment:
What's the motivation for making this a generic abstraction called Size?

Are there any drawbacks to explicitly listing the possible values in each component that has a size prop?

Reply:
If it's an abstraction, is there a name we could use that is more specific to its use, such as `ComponentSizeVariant`?

Reply:
@penx I noticed we use below in some components. As the sizes are shared it would be really good to start combining in one common place. It would also mean the sizes can be consistent. Is that ok?

I will update:
- name to be `ComponentSizeVariant`
- Values to match existing values: `XLARGE`, `LARGE`, `MEDIUM`, `SMALL`

```
`XLARGE`, `LARGE`, `MEDIUM`, `SMALL`, `XL`, `L`, `M`, `S`

Reply:
yes, agreed it would be good to make sure the strings we pass in to the sizeVariant props across components are consistent, so I agree with the abstraction for that